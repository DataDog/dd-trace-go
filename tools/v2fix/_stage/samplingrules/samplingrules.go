// Unless explicitly stated otherwise all files in this repository are licensed
// under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2024 Datadog, Inc.

package main

import (
	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
)

func tags() map[string]string {
	return map[string]string{"hostname": "hn-*", "another": "value"}
}

func main() {
	_ = tracer.ServiceRule("test-service", 1.0)                                                // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.NameRule("http.request", 1.0)                                                   // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.NameServiceRule("http.request", "test-service", 1.0)                            // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.NameServiceRule("http.*", "test-*", 1.0)                                        // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.ServiceRule("other-service-1", 0.0)                                             // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.ServiceRule("other-service-2", 0.0)                                             // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.ServiceRule("test-service", 1.0)                                                // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.TagsResourceRule(tags(), "", "", "", 1.0)                                       // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.TagsResourceRule(map[string]string{"hostname": "hn-3*"}, "res-1*", "", "", 1.0) // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.TagsResourceRule(map[string]string{"hostname": "hn-*"}, "", "", "", 1.0)        // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
}
