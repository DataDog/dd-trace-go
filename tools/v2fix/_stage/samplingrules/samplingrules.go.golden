-- a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal --
// Unless explicitly stated otherwise all files in this repository are licensed
// under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2024 Datadog, Inc.

package main

import (
	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
)

func tags() map[string]string {
	return map[string]string{"hostname": "hn-*", "another": "value"}
}

func main() {
	_ = tracer.TraceSamplingRules(Rule{ServiceGlob: "test-service", Rate: 1.0})                                                                         // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.TraceSamplingRules(Rule{NameGlob: "http.request", Rate: 1.0})                                                                            // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.TraceSamplingRules(Rule{NameGlob: "http.request", ServiceGlob: "test-service", Rate: 1.0})                                               // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.TraceSamplingRules(Rule{NameGlob: "http.*", ServiceGlob: "test-*", Rate: 1.0})                                                           // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.TraceSamplingRules(Rule{ServiceGlob: "other-service-1", Rate: 0.0})                                                                      // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.TraceSamplingRules(Rule{ServiceGlob: "other-service-2", Rate: 0.0})                                                                      // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.TraceSamplingRules(Rule{ServiceGlob: "test-service", Rate: 1.0})                                                                         // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.TraceSamplingRules(Rule{Tags: tags(), ResourceGlob: "", NameGlob: "", ServiceGlob: "", Rate: 1.0})                                       // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.TraceSamplingRules(Rule{Tags: map[string]string{"hostname": "hn-3*"}, ResourceGlob: "res-1*", NameGlob: "", ServiceGlob: "", Rate: 1.0}) // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.TraceSamplingRules(Rule{Tags: map[string]string{"hostname": "hn-*"}, ResourceGlob: "", NameGlob: "", ServiceGlob: "", Rate: 1.0})        // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.SpanSamplingRules(Rule{NameGlob: "http.request", ServiceGlob: "test-service", Rate: 1.0, MaxPerSecond: 2.0})                             // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.SpanSamplingRules(Rule{Tags: tags(), ResourceGlob: "", NameGlob: "", ServiceGlob: "", Rate: 1.0})                                        // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
	_ = tracer.SpanSamplingRules(Rule{NameGlob: "http.request", ServiceGlob: "test-service", Rate: 1.0})                                                // want `a deprecated sampling rule constructor function should be replaced with a tracer.Rule{...} struct literal`
}
